File: 004Fibonacci-Swap-MinMax/Python/fib.py

def fib(last_num):
    num1 = 1
    num2 = 0

    if last_num < 0:
        return

    print(num2)
    while num1 <= last_num:
        print(num1)
        num1, num2 = num1+num2, num1

    
if __name__ == '__main__':
    fib(10000)

File: 004Fibonacci-Swap-MinMax/CSharp/fib.cs


class Fib{

    static void Main(string[] args)
    {
        fib(10000);
    }

    static void fib(int last_num)
    {
        int num1=1, num2=0,tmp;
        if(num2 > last_num) return;

        System.Console.WriteLine(num2);
        while(num1 < last_num)
        {
            System.Console.WriteLine(num1);
            tmp = num1 + num2;
            num2 = num1;
            num1 = tmp;
        }
    }
}
